modules/videoio/test/test_video_io.cpp

```diff
@@ -171,8 +177,9 @@ class Videoio_Bunny : public Videoio_Test_Base, public testing::TestWithParam<st
         else
             std::cout << "FPS is not available. SKIP check." << std::endl;
 
-        int count_prop = (int)cap.get(CAP_PROP_FRAME_COUNT);
-
+        int count_prop = 0;
+#ifndef HAVE_AVFOUNDATION
```

Comment:
This check should be replaced with dynamic check, because otherwise this part of the test will be disabled for all backends on Mac.
```.cpp
if (apiPref == CAP_AVFOUNDATION)
{
    // TODO: fix this backend
    std::cout << "SKIP test: AVFoundation backend returns invalid frame count" << std::endl;
    return;
}